### PR TITLE
Add support for sending POST requests as queries

### PR DESCRIPTION
**Pull request by OverloadUT**
POST queries require a client identifier

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -146,6 +163,52 @@ PlexAPI.prototype._request = function _request(relativeUrl) {
     return deferred.promise;
 };
 
+PlexAPI.prototype._post = function perform(relativeUrl) {
```
Comment (phillipj) [inline lib/api.js]: `_post()` looks very similar to `_request()`. Maybe `_request()` could be refactored to handle both GET and POST?

File: lib/api.js [hunk 2]
```diff
@@ -153,6 +216,11 @@ PlexAPI.prototype._authenticateAndRetry = function _authenticateAndRetry(url, re
         return Q.reject(new Error('Permission denied even after attempted authentication :( Wrong username and/or password maybe?'));
     }
 
+    // Bit of a kludgey way to generate an Identifier before we send off the options to auth.
+    // Right now it doesn't appear that using the same identifier on every request is necessary, but
+    // it's probably good practice to do so.
+    self.getIdentifier();
```
Comment (phillipj) [inline lib/api.js]: As this ensures an identifier is always generated, should it be moved into the constructor instead? Something like

```js
this.options.identifier = this.options.identifier || uuid.v4();
```
on line #21?

**Comment by phillipj** [comment]
Thanks for more contributions @OverloadUT!

A couple of tests covering POST requests + documenting these two methods you have added in the readme would have been nice :smiley:

**Comment by OverloadUT** [comment]
I will add those! I had meant to, but totally forgot. I'm fixing the code formatting issues that Travis pointed out as well.

**Comment by OverloadUT** [comment]
Alright, I did a big refactoring for this PR. When combining _request and _post, it just didn't make sense to also have perform() there doing 90% of the same stuff _request does. So I moved it so that all three kinds of requests (other than auth) go through _request.

Also added tests and documentation :)

**Comment by phillipj** [comment]
Wow, great refactoring @OverloadUT! I'll merge and publish v2.4.0 right away :smiley:

Would you like to be added as a collaborator of this project?

**Comment by OverloadUT** [comment]
I'd be happy to be added!

I've got one more PR coming up that will then catch it up to my "xml2js instead of xml2json" fork that I'm using.

**Comment by phillipj** [comment]
There you go, welcome onboard :)

What's the benefit of using xml2js compared to xml2json?

**Comment by OverloadUT** [comment]
The tl;dr is that I need it to not require a C library.

It's a very specific requirement in my case: I am deploying my code to **AWS Lambda**. The way that works is I need to upload a zip of the project which can run on Amazon's linux distro. The problem is that I'm developing on Windows, so when I ```npm install``` my module and upload, it ends up with a Windows-compiled C library which of course doesn't work in the linux environment. (It also makes the deploy package massively larger, but that doesn't *really* matter)

The way to really fix it on my side is to have an Amazon Linux VM and do my deploys from there. I might set that up one of these days, but for now it's been much easier to simply change two lines of code in node-plex-api.

**Comment by phillipj** [comment]
Getting rid of the native addon sounds like a very good idea. Especially
when there's no appearent reason to use native. I've spent a good deal of
hours debugging such modules, not always trivial for non-native-ninjas.
